methods.py

```diff
@@ -650,39 +657,112 @@ def add_to_vs_project(env, sources):
 
 
 def generate_vs_project(env, num_jobs):
+
```

Comment:
I'm surprised that the `black` formatter doesn't complain here (might be broken in CI), but empty lines at the start of functions should be removed.